File: services/screenshot.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
from playwright.sync_api import sync_playwright
import os
import time

def take_screenshot(html_path, screenshot_path):
    os.makedirs(os.path.dirname(screenshot_path), exist_ok=True)

    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()

        with open(html_path, 'r', encoding='utf-8') as f:
            html_content = f.read()

        page.set_content(html_content, wait_until="networkidle")
        try:
            page.wait_for_selector('.fa-bar-chart', timeout=5000)
            page.click('.fa-bar-chart')
            time.sleep(1)
        except Exception as e:
            print(f"⚠️ Element not found or failed: {e}")

        page.set_viewport_size({"width": 1920, "height": 1080})
        page.screenshot(path=screenshot_path, full_page=True)
        print(f" Screenshot saved: {screenshot_path}")
        browser.close()

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: screenshot.py <html_path> <screenshot_path>")
        sys.exit(1)
    take_screenshot(sys.argv[1], sys.argv[2])

File: services/screenshot.js
// const { chromium } = require('playwright');
// const fs = require('fs');
// const path = require('path');

// async function takeScreenshot(htmlPath, screenshotPath) {
//   const delay = ms => new Promise(resolve => setTimeout(resolve, ms));

//   // Ensure folder exists
//   const screenshotDir = path.dirname(screenshotPath);
//   if (!fs.existsSync(screenshotDir)) {
//     fs.mkdirSync(screenshotDir, { recursive: true });
//     console.log(`📁 Created folder: ${screenshotDir}`);
//   }

//   const browser = await chromium.launch(); // No need for executablePath
//   const page = await browser.newPage();

//   const htmlContent = fs.readFileSync(htmlPath, 'utf8');
//   await page.setContent(htmlContent, { waitUntil: 'networkidle' });

//   try {
//     await page.waitForSelector('.fa-bar-chart', { timeout: 5000 });
//     await page.click('.fa-bar-chart');
//   } catch (err) {
//     console.warn('⚠️ .fa-bar-chart not found or clickable:', err.message);
//   }

//   await delay(1000);
//   await page.setViewportSize({ width: 1920, height: 1080 });
//   await page.screenshot({ path: screenshotPath, fullPage: true });

//   console.log(`📸 Screenshot saved: ${screenshotPath}`);
//   await browser.close();
//   console.log('🧹 Playwright browser closed');
// }

// module.exports = { takeScreenshot };

const { spawn } = require('child_process');
const path = require('path');

function take_screenshot(htmlPath, screenshotPath) {
  return new Promise((resolve, reject) => {
    const pythonScript = path.join(__dirname, '../services/screenshot.py');
    const pythonCmd = process.platform === 'win32' ? 'python' : 'python3';

    const proc = spawn(pythonCmd, [pythonScript, htmlPath, screenshotPath]);
    let stderr = '';

    proc.stderr.on('data', data => {
      stderr += data.toString();
    });

    proc.on('close', code => {
      if (code === 0) {
        resolve();
      } else {
        reject(new Error(`Screenshot failed: ${stderr}`));
      }
    });
  });
}

module.exports = { take_screenshot };